examples/all-devices-app/linux/main.cpp

```diff
@@ -0,0 +1,283 @@
+#include <app/clusters/software-diagnostics-server/software-diagnostics-cluster.h>
+#include <app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-cluster.h>
+#include <app/persistence/DefaultAttributePersistenceProvider.h>
+#include <app/server-cluster/ServerClusterInterfaceRegistry.h>
+#include <app/server/Dnssd.h>
+#include <app/server/Server.h>
+#include <credentials/examples/DeviceAttestationCredsExample.h>
+#include <data-model-providers/codedriven/CodeDrivenDataModelProvider.h>
+#include <data-model-providers/codedriven/endpoint/SpanEndpoint.h>
+#include <devices/contact-sensor/ContactSensorDevice.h>
+#include <devices/device-factory/DeviceFactory.h>
+#include <devices/root-node/RootNodeDevice.h>
+#include <platform/CommissionableDataProvider.h>
+#include <platform/Linux/NetworkCommissioningDriver.h>
+#include <platform/PlatformManager.h>
+#include <setup_payload/OnboardingCodesUtil.h>
+#include <system/SystemLayer.h>
+
+#include <AppMain.h>
+#include <map>
+#include <string>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::Platform;
+using namespace chip::DeviceLayer;
+using namespace chip::app::Clusters;
+using namespace chip::ArgParser;
+
+namespace {
+AppMainLoopImplementation * gMainLoopImplementation = nullptr;
+
+DeviceLayer::NetworkCommissioning::LinuxWiFiDriver sWiFiDriver;
+
+DeviceInfoProviderImpl gExampleDeviceInfoProvider;
+
+// To hold SPAKE2+ verifier, discriminator, passcode
+LinuxCommissionableDataProvider gCommissionableDataProvider;
+
+// App custom argument handling
+constexpr uint16_t kOptionDeviceType = 0xffd0;
+const char * deviceTypeName          = "contact-sensor"; // defaulting to contact sensor if not specified
+
+chip::ArgParser::OptionDef sAllDevicesAppOptionDefs[] = {
+    { "device", chip::ArgParser::kArgumentRequired, kOptionDeviceType },
+};
+
+bool AllDevicesAppOptionHandler(const char * program, OptionSet * options, int identifier, const char * name, const char * value)
+{
+    switch (identifier)
+    {
+    case kOptionDeviceType:
+        if (value == nullptr || !DeviceFactory::GetInstance().IsValidDevice(value))
+        {
+            ChipLogError(Support, "INTERNAL ERROR: Invalid device type: %s\n", value);
+            return false;
+        }
+        ChipLogProgress(AppServer, "Using the device type of %s", value);
+        deviceTypeName = value;
+        return true;
+    default:
+        ChipLogError(Support, "%s: INTERNAL ERROR: Unhandled option: %s\n", program, name);
+        return false;
+    }
+
+    return true;
+}
+
+chip::ArgParser::OptionSet sCmdLineOptions = { AllDevicesAppOptionHandler, // handler function
+                                               sAllDevicesAppOptionDefs,   // array of option definitions
+                                               "PROGRAM OPTIONS",          // help group
+                                               "-d, --device <contact-sensor|water-leak-detector>\n" };
```

Comment:
we may need a way to scale this ... this comment would become obsolete fast i imagine. Not needed for this PR, but maybe for the future.